### PR TITLE
[fpga] Enable splicing OTP images into bitstream

### PR DESCRIPTION
**Pull request by dmcardle**
* Generate otp.mmi in vivado_hook_write_bitstream_pre.tcl
* Add otp.mmi to outputs of //hw/bitstream/vivado:fpga_cw310
* Reuse bitstream_splice() macro to splice in the dev OTP image after
  splicing in the Mask ROM.

https://github.com/lowRISC/opentitan/issues/10867

### CODE REVIEW

**Review by cfrantz — APPROVED** [review]
The bazel changes LGTM.  I'll let msf@ comment on the TCL changes.

File: hw/top_earlgrey/util/vivado_hook_write_bitstream_pre.tcl [hunk 1]
```diff
@@ -16,67 +16,76 @@ if [expr {$slack_ns < 0}] {
 set_property BITSTREAM.CONFIG.USR_ACCESS TIMESTAMP [current_design]
 
 # Dump MMI for Boot ROM.
-send_msg "Designtask 1-1" INFO "Dumping Boot ROM MMI"
```
[the remark this answers is not in the document]
Comment (dmcardle) [reply hw/top_earlgrey/util/vivado_hook_write_bitstream_pre.tcl:19]: Oh, it also modified the send_msg lines, but I figured those are purely informative.

File: sw/device/silicon_creator/mask_rom/BUILD [hunk 2]
```diff
@@ -329,6 +329,24 @@ opentitan_functest(
     ],
 )
 
+# Same as `:e2e_bootup_success`, but the Dev OTP image is spliced into the
+# bitstream before it's sent to the CW310 FPGA.
+opentitan_functest(
+    name = "e2e_bootup_success_otp_dev",
```
[the remark this answers is not in the document]
Comment (dmcardle) [reply sw/device/silicon_creator/mask_rom/BUILD:338]: Just pushed 7fe7443e35894b1c52659e7e8bad9baf4b77d5d0 to see whether the "manual" tag was necessary.

[the remark this answers is not in the document]
Comment (dmcardle) [reply sw/device/silicon_creator/mask_rom/BUILD:338]: Hooray, it works. Squashing into the previous commit.

**Review by msfschaffner — APPROVED** [review]
This LGTM from my end!

**Comment by dmcardle** [comment]
Woohoo, thanks!

**Comment by alphan** [comment]
Nice!

**Pull request merged**